### ISSUE TITLE
Add warning in doc generator when doc comment is overriden

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#8338]
When types are reopened, there can be multiple doc comments for the same thing:
```cr
# Foo
class Foo
end

# Foo class
class Foo
end
```
Only one of them is used by the docs generator (I think it's the latter?) and the other is simply overridden. This is not desired behaviour. It doesn't make sense to simply concatenate the docs either. There should ever be only a single place for documentation a single feature.
So when generating docs, the compiler could show a warning when doc has been defined before. Alternatively, it could directly fail, but I think a warning is fine.

This would avoid issues like #8337

**Comment by asterite** [comment]
Yes! I had the same idea in the past but I forgot to turn it into an issue or PR.

**Comment by RX14** [comment]
It should either fail or override. Compiler warnings should be avoided.

**Comment by straight-shoota** [comment]
@RX14 What about deprecation warnings?

We could deprecate this at first and make it fail in the next release.
I'm fine with failing directly, though. It's trivial to fix so won't be a big hold up when doc generation fails because of this.

We should however decide whether doc errors like this should only fail the doc generator or also the compiler (i.e. should it be a syntax error?). I don't think this needs to affect the compiler. Programs with documentation errors should still be able to compile.

**Comment by RX14** [comment]
Deprecations are the only kind of warnings the compiler should provide, imo. I've been working to make them less verbose (I've stated my vision for the deprecations UX many times before) too but the code is a quagmire and unfortunately I'm really busy ;_;

Ary said a long time ago that crystal didn't have compiler warnings because either the feature had a usecase - or it should be explicitly disallowed with a compiler error. Linters exist to fill the usecase of opinionated "this isn't best practice" - not the compiler. I still believe in that vision for how warnings should be treated by us.

**Comment by bcardiff** [comment]
I think in this case we are discussing a warning / output in the doc generator tool. And not in the compiler itself.

Something like:

> hey! you might want to add a blank line if you don't want the comment to be a doc

could be a good suggestion.

**Comment by straight-shoota** [comment]
I'd prefer to make overriding documentation a failure, either only for the doc generator or maybe even a generic syntax error.

**Comment by nobodywasishere** [comment]
Could we have the docs just merged together in the order they're parsed?

**Comment by straight-shoota** [comment]
That would be pretty confusing and have weird effects.
There should ever be only a single place for documentating a single feature.
